lib/rules/comma-dangle.js

```diff
@@ -13,14 +13,17 @@ const lodash = require("lodash");
 
 /**
  * Checks whether or not a trailing comma is allowed in a given node.
- * `ArrayPattern` which has `RestElement` disallows it.
+ * `ArrayPattern` which has `RestElement` or `ObjectPattern` which has `RestProperty` disallows it.
  *
  * @param {ASTNode} node - A node to check.
  * @param {ASTNode} lastItem - The node of the last element in the given node.
  * @returns {boolean} `true` if a trailing comma is allowed.
  */
 function isTrailingCommaAllowed(node, lastItem) {
-    return node.type !== "ArrayPattern" || lastItem.type !== "RestElement";
+    return !(
+        (node.type === "ArrayPattern" && lastItem.type === "RestElement") ||
+        (node.type === "ObjectPattern" && (lastItem.type === "RestProperty" || lastItem.type === "ExperimentalRestProperty"))
```

Comment:
Espree uses `ExperimentalRestProperty` because the feature was experimental. If there was a change before the proposal was fully added to the specification (which just happened), we had the ability to ensure that rules could target both the old representation and the official versions (`RestProperty`). I didn't want people relying on an experimental node without knowing that they were doing so.

When we fully implement the spec, we'll use `RestProperty`.